app/src/main/java/com/example/playlistmaker/ui/activity/PlayerActivity.kt

```diff
@@ -1,28 +1,55 @@
 package com.example.playlistmaker.ui.activity
 
+import android.media.MediaPlayer
 import android.os.Bundle
-import android.util.Log
+import android.os.Handler
 import android.view.View.GONE
+import android.widget.ImageButton
 import android.widget.ImageView
 import android.widget.TextView
 import androidx.appcompat.app.AppCompatActivity
 import androidx.constraintlayout.widget.Group
+import androidx.core.content.IntentCompat
 import com.bumptech.glide.Glide
 import com.bumptech.glide.load.resource.bitmap.RoundedCorners
 import com.example.playlistmaker.R
 import com.example.playlistmaker.data.Track
+import com.example.playlistmaker.ui.PlayerState.PAUSED
+import com.example.playlistmaker.ui.PlayerState.PLAYING
+import com.example.playlistmaker.ui.PlayerState.PREPARING
+import com.example.playlistmaker.ui.PlayerState.STOPPED
 import com.example.playlistmaker.ui.activity.SearchActivity.Companion.PLAY_TRACK
 import com.example.playlistmaker.ui.recyclerVIew.TrackViewHolder.Companion.simpleDateFormat
 import com.example.playlistmaker.utils.transformDpToPx
-import com.google.gson.Gson
+import java.text.SimpleDateFormat
+import java.util.Locale
 
 class PlayerActivity : AppCompatActivity() {
     private var currentTrack: Track? = null
+    private val player by lazy { MediaPlayer() }
+
+    private var playerState = STOPPED
+    private var isButtonPlayClicked = false
+    private var isTrackStoppedBeforeOnPause = false
+
+    private lateinit var playerControlButton: ImageButton
+    private lateinit var currentTrackTime: TextView
+
+    private val timeControlRunnable = object : Runnable {
+        override fun run() {
+            currentTrackTime.text =
+                SimpleDateFormat("mm:ss", Locale.getDefault()).format(player.currentPosition)
+            mainHandler.postDelayed(this, UPDATE_CURRENT_TIME_DELAY)
+        }
+    }
+    private val mainHandler by lazy { Handler(mainLooper) }
```

Comment:
Синхронизацию можно убрать:
```suggestion
    private val mainHandler by lazy(mode = LazyThreadSafetyMode.NONE) { Handler(mainLooper) }
```